### ISSUE TITLE
Export error with saved_model, pb, tflite 

### ISSUE THREAD

**Issue by ShubhamAtPhilips** [ultralytics/yolov5#7091]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar bug report.

### YOLOv5 Component

Export

### Bug

I trained a model using this [models/hub/yolov3-tiny.yaml](https://github.com/ultralytics/yolov5/blob/master/models/hub/yolov3-tiny.yaml ) configuration. I am trying to convert it to TFlite. 

I found out that I need to implement TFMaxPool2d and TFZeroPad2d in [models/tf.py](https://github.com/ultralytics/yolov5/blob/master/models/tf.py) from this issue https://github.com/ultralytics/yolov5/issues/7033.

I implemented it as follows in [models/tf.py](https://github.com/ultralytics/yolov5/blob/master/models/tf.py)

```
class TFMaxPool2d(keras.layers.Layer):
    def __init__(self, k=(2, 2), s=None, p='same', w=None):
        super().__init__()
        self.m = keras.layers.MaxPool2D(pool_size=k, strides=s, padding='SAME')

    def call(self, inputs):
        return self.m(inputs)

class TFZeroPad2d(keras.layers.Layer):
    def __init__(self, p=(1, 1), w=None):
        super().__init__()
        print(p)
        self.z = keras.layers.ZeroPadding2D(padding=p)

    def call(self, inputs):
        return self.z(inputs)
```
Now, I am getting the error 

```
TensorFlow SavedModel: starting export with tensorflow 2.7.1...
                 from  n    params  module                                  arguments                        0                -1  1       464  models.common.Conv                      [3, 16, 3, 1]                    1                -1  1         0  torch.nn.modules.pooling.MaxPool2d      [2, 2, 0]                        2                -1  1      4672  models.common.Conv                      [16, 32, 3, 1]                   3                -1  1         0  torch.nn.modules.pooling.MaxPool2d      [2, 2, 0]                        4                -1  1     18560  models.common.Conv                      [32, 64, 3, 1]                   5                -1  1         0  torch.nn.modules.pooling.MaxPool2d      [2, 2, 0]                        6                -1  1     73984  models.common.Conv                      [64, 128, 3, 1]                  7                -1  1         0  torch.nn.modules.pooling.MaxPool2d      [2, 2, 0]                        8                -1  1    295424  models.common.Conv                      [128, 256, 3, 1]                 9                -1  1         0  torch.nn.modules.pooling.MaxPool2d      [2, 2, 0]                       10                -1  1   1180672  models.common.Conv                      [256, 512, 3, 1]               

TensorFlow SavedModel: export failure: `padding` should have two elements. Received: [0, 1, 
0, 1].

TensorFlow Lite: starting export with tensorflow 2.7.1...

TensorFlow Lite: export failure: 'NoneType' object has no attribute 'call'
```

Can someone help me implement 4-tuple padding TF/Keras equivalent of [torch.nn.ZeroPad2d](https://pytorch.org/docs/stable/generated/torch.nn.ZeroPad2d.html)? 

Thank you.

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by ShubhamAtPhilips** [comment]
Update,

I updated the function definitions, and it seems to work now. 

```
class TFMaxPool2d(keras.layers.Layer):
    def __init__(self, k=(2, 2), s=None, p='same', w=None):
        super().__init__()
        if p == 0:
            padding = 'valid'
        else:
            padding = 'same'
        self.m = keras.layers.MaxPool2D(pool_size=k, strides=s, padding=padding)
        
    def call(self, inputs):
        return self.m(inputs)

class TFZeroPad2d(keras.layers.Layer):
    def __init__(self, p=(1, 1), w=None):
        super().__init__()
        if hasattr(p, '__len__'):
            if len(p) == 4:
                p = ((p[2], p[3]), (p[0], p[1]))
        self.z = keras.layers.ZeroPadding2D(padding=p)

    def call(self, inputs):
        return self.z(inputs)
```

Can someone verify if it's correct? If yes, I can submit a PR.

**Comment by glenn-jocher** [comment]
@ShubhamAtPhilips oh cool! Yes go ahead and submit a PR. To verify your implementation export yolov3-tiny.pt to TFLite or SavedModel format and then run val.py on your exported model and compare against PyTorch model. Both should have identical mAP to several decimal places.

**Comment by wakasturner** [comment]
> 

This works, i was able to convert yolov3-tiny to tflite 👍

**Comment by glenn-jocher** [comment]
@wakasu that's great to hear! 🎉 If you have any further questions or need assistance with anything else, feel free to ask. Good luck with your project!
